lib/iolog/log_reader.go

```diff
@@ -35,68 +35,102 @@ type LogReader struct {
 	ts   bool
 }
 
+// Header describes the header of a containerVM log entry
+type Header struct {
+	Timestamp time.Time
+	Size      int
+	Stream    int
+}
+
+// Entry describes a containerVM log entry
+type Entry struct {
+	Header
+	Message []byte
+}
+
 // NewLogReader wraps an io.ReadCloser in a LogReader.
 func NewLogReader(r io.ReadCloser, ts bool) *LogReader {
 	return &LogReader{
 		ReadCloser: r,
 		ts:         ts}
 }
 
+// ParseLogEntry parses data from an io.Reader into a Entry
+func ParseLogEntry(r io.Reader) (*Entry, error) {
+	var (
+		err  error
+		n, w int
+		ts   time.Time
+	)
+
+	ehdr := make([]byte, encodedHeaderLengthBytes)
+	// read a header
+	n = 0
+	for n < encodedHeaderLengthBytes {
+		w, err = r.Read(ehdr[n:])
+		n += w
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// decode base64 header
+	hdr, err := base64.StdEncoding.DecodeString(string(ehdr))
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
I don't think its necessary, since they are always exactly 16 bytes. If we don't somehow get the whole thing in the previous `Read`, we 
 would (in theory) error during that `for n < encodedHeaderLengthBytes` loop, and should also get an error attempting to base64 decode.